docs/design/enforceHostNetwork.md

```diff
@@ -0,0 +1,32 @@
+# Enforcing host network design document
+
+By default, the ceph-csi controller plugin operates on the pod network
+but under some circumstances, like like setups with a dedicaated storage network, 
+where the pod network cannot connect to the ceph cluster,
+it is necessary 
+to run the Ceph-CSI controller plugin pods on the host network.
+
+This document describes how the ceph-csi-operator can be configured to enforce the use of host network in 
+the Ceph-CSI controller plugin pods
+
+Enforcing of host network can be enabled in the operator configuration by setting `hostNetwork` to `true` in the `driverSpecDefaults.rcontrollerPlugin` section
+of the `OperatorConfig` CR.
+
+Example:
+
+```yaml
+kind: OperatorConfig 
+apiVersion: csi.ceph.io/v1alpha1
+
+
+metadata:
+  name: ceph-csi-operator-config
+  namespace: <operator-namespace>
+spec:
+  driverSpecDefaults:
+     controllerPlugin:
+         hostNetwork: true
```

Comment:
Would suggest to rename this to something like `useHostNetwork`

Reply:
> Would suggest to rename this to something like `useHostNetwork`

Originally, I had `EnforceHostNetwork`, but @Madhu-1  asked me to change that to `hostNetwork`  --  So what now? 😬

Reply:
done